template/template.go

```diff
@@ -249,10 +319,10 @@ func (e *executer) resolvePattern(pattern []ast.PatternPart) error {
 	return resolutionErr
 }
 
-func (e *executer) resolveExpression(expr ast.Expression) (string, error) {
+func (e *executer) resolveExpression(expr ast.Expression) (*ResolvedValue, error) {
```

Comment:
Why does the consumer of `resolveExpression()` handle fallback value? Why doesn't `resolveExpression` return a fallback value?

Reply:
https://github.com/translate-agent/mf2/pull/191/files#diff-d24cab31570f37b8fb2190891c0a53a42e2243d6d242df62c2ef59a7f9d8c972R278

Reply:
Because the expression sometimes has no information about the variable it was meant to bind to, e.g. in `.input` declarations.

Reply:
`.input {$n :number} {{Count: $n}}`. What's missing in `{$n :number}`? Isn't fallback value `{$n}`?

Reply:
Currently the declaration resolving replaces the `$n` with the value from external input parameters to avoid circular reference problems. Maybe it's wise to keep declarations in a separate map from the external input params. I'll see if that solves problems.

Reply:
No need for a separate map, refactoring solved the problem.